oteapi/models/secretconfig.py

```diff
@@ -0,0 +1,67 @@
+"""AttrDict for specifying user credentials or secrets."""
+
+import json
+from typing import Optional
+
+from pydantic import Field, SecretStr
+
+from oteapi.models.genericconfig import AttrDict
+from oteapi.settings import settings
+
+
+def json_dumps(model, *, default):
+    """Alternative function for dumping exposed
+    secrets to json when model is serialized."""
```

Comment:
```suggestion
def json_dumps(model: SecretConfig, *, default: "Callable[[Any], Any]") -> "Dict[str, Any]":
    """Alternative function for dumping exposed
    secrets to json when model is serialized.

    Parameters:
        model: The pydantic model to serialize.
        default: A pass-through to the standard `json.dumps()`'s `default` parameter.
            From the `json.dumps()` doc-string: `default(obj)` is a function that should
            return a serializable version of `obj` or raise `TypeError`.
            The default simply raises `TypeError`.

    Returns:
        The result of `json.dumps()` after handling possible secrets.

    """
```
Two things:

1. Updated the signature to include types.
   For this, some classes needs to be imported from `typing`.
2. Updated the doc-string to be more informative about the parameters.